File: src/test/scala/com/exasol/cloudetl/scriptclasses/ExportPathTest.scala
package com.exasol.cloudetl.scriptclasses

import scala.collection.JavaConverters._

import org.mockito.Mockito._

class ExportPathTest extends PathTest {

  test("generateSqlForExportSpec throws with message to the new scriptclass name") {
    when(metadata.getScriptSchema()).thenReturn(schema)
    when(exportSpec.getParameters()).thenReturn(properties.asJava)
    val thrown = intercept[IllegalArgumentException] {
      ExportPath.generateSqlForExportSpec(metadata, exportSpec)
    }
    assert(thrown.getMessage().contains("Please use the ExportTableQueryGenerator"))
  }

}

File: src/test/java/com/exasol/cloudetl/scriptclasses/ExportPathTest.java
package com.exasol.cloudetl.scriptclasses;

import static org.junit.jupiter.api.Assertions.*;
import static org.mockito.Mockito.*;

import org.junit.jupiter.api.Test;

class ExportPathTest extends PathTest {
    @Test
    void generateSqlForExportSpecThrowsWithMessageToTheNewScriptclassName() {
        when(this.metadata.getScriptSchema()).thenReturn(this.schema);
        when(this.exportSpec.getParameters()).thenReturn(this.properties);
        final IllegalArgumentException thrown = assertThrows(IllegalArgumentException.class,
                () -> ExportPath.generateSqlForExportSpec(this.metadata, this.exportSpec));
        assertTrue(thrown.getMessage().contains("Please use the TableExportQueryGenerator"));
    }
}
